src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
-
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.health = 500;
+    this.name = 'Test';
+    Animal.alive.push(this.name);
+  }
+  static deleteIndex = Animal.alive.findIndex((animal) => animal.health !== 'dead');
+
+  deleteAnimal() {
+    Animal.alive = Animal.alive.filter((animal) => animal.name = 'dad');
```

Comment:
The filter callback is incorrect; it should compare the animal's name to the current instance's name (this.name) instead of setting it to 'dad'. Also, the filter should not be used for deleting a single item; consider using splice with the correct index.